test/integration/iterate_by_column_test.rb

```diff
@@ -0,0 +1,118 @@
+
+  def test_specifying_pagination_column_by_fallback_column_succeeds
+    pagination_key_column = 'some_column'
+    db_name = DEFAULT_DB
+    table_name = DEFAULT_TABLE
+
+    create_tables(db_name, table_name, "(#{pagination_key_column} bigint(20), data BLOB)")
+    seed_dynamically(DbHelper.full_table_name(db_name, table_name), {
+      pagination_key_column => lambda {|i| i},
+      data: lambda {|_| "'some data'"}
+    }, amount_rows: 10)
+
+    assert_ghostferry_run(fallback_column: pagination_key_column)
+    assert_source_target_eql_data(DbHelper.full_table_name(db_name, table_name))
+  end
+
+  def test_specifying_pagination_column_by_per_table_and_fallback_column_succeeds
+    pagination_key_column_1 = 'some_column'
+    pagination_key_column_2 = 'other_column'
+
+    db_name = DEFAULT_DB
+    table_name_1 = DEFAULT_TABLE
+    table_name_2 = "other_table"
+
+    create_tables(db_name, table_name_2, "(#{pagination_key_column_2} bigint(20), data BLOB)")
+    # Following creates composite index: other_id, some_column
+    create_tables(db_name, table_name_1, "(#{pagination_key_column_1} bigint(20), other_id bigint(20), data BLOB, primary key(other_id, #{pagination_key_column_1}))")
+
+
+    seed_dynamically(DbHelper.full_table_name(db_name, table_name_1), {
+      pagination_key_column_1 => lambda {|i| i},
+      other_id: lambda {|i| i},
+      data: lambda {|_| "'some data'"}
+    }, amount_rows: 10)
+
+    seed_dynamically(DbHelper.full_table_name(db_name, table_name_2), {
+      pagination_key_column_2 => lambda {|i| i},
+      data: lambda {|_| "'some data'"}
+    }, amount_rows: 10)
+
+    assert_ghostferry_run(per_table: {
+      db_name => {
+        table_name_1 => pagination_key_column_1,
+      },
+    }, fallback_column: pagination_key_column_2)
+
+    assert_source_target_eql_data(DbHelper.full_table_name(db_name, table_name_1))
+    assert_source_target_eql_data(DbHelper.full_table_name(db_name, table_name_2))
+  end
+
+  private
+  def create_tables(db_name, table_name, create_query)
+    [source_db, target_db].each do |db|
+      db.query("CREATE DATABASE IF NOT EXISTS #{db_name}")
+      db.query("CREATE TABLE IF NOT EXISTS #{DbHelper.full_table_name(db_name, table_name)} #{create_query}")
+    end
+  end
+
+  def seed_dynamically(full_table_name, columns, amount_rows:)
+    (1..amount_rows).map do |i|
+      column_names = columns.keys.join(",")
+      column_values = columns.values.map {|l| l.(i)}.join(",")
+
+      "(#{column_names}) VALUES (#{column_values})"
+    end.each do |insert_query_portion|
+      source_db.query("INSERT INTO #{full_table_name} #{insert_query_portion}")
+    end
+  end
+
+  def cascadingPaginationColumnConfig(per_table: nil, fallback_column: nil)
+    {
+      PerTable: per_table,
+      FallbackColumn: fallback_column
+    }.reduce({}) do |a,(k,v)|
+      if v
+        a.merge!({k => v})
+      end
+      a
+    end.to_json
+  end
+
+  def assert_source_target_eql_data(full_table_name)
```

Comment:
This has been introduced as https://github.com/Shopify/ghostferry/blob/85f2a57620c9580d280162c9081aeb27dc822fda/test/test_helper.rb#L84 requires a bit extra customization, in terms of a parameter for a different table(which is trivial) and in terms of the specificity of: https://github.com/Shopify/ghostferry/blob/b5a11f7f8b8ae58d8811653e3a310e0bdcd2e87f/test/helpers/db_helper.rb#L126 which has other tests depending on it